### DISCUSSION TITLE
v6: Using a non-default schema with SQL Server leads to error on create adonis schema tables

### DISCUSSION THREAD

**Discussion by jsilva74** [adonisjs/core#4531] [Help]
Hello everybody.

I have this project which uses a SQL Server database and should use a schema other than dbo.

I've successfully created the migrations and have setted up the tableName in config/database.ts to `[app].[adonis_schema]`, pointing to the correct schema but, when tried to run the migrations, I'm getting an error due to the constraint name in field `migration_time` (the constraint name is using the tableName as is - `[app].[adonis_schema_migration_time_default]` - causing the error because of the `.` (separator between the schema and table name).

Tried and can't found a way to go around this. If anyone could help, will be deeply appreciated. Thanks.

**Comment by Steveperi** [comment]
Hello, 
Have you tried this?
```
import { BaseSchema } from '@adonisjs/lucid/schema'

export default class extends BaseSchema {
  protected tableName = 'table'

  async up() {
    this.schema.withSchema('schema').createTable(this.tableName, (table) => {
      table.string('name', 255).notNullable().unique()
    })
  }

  async down() {
    this.schema.withSchema('schema').dropTable(this.tableName)
  }
}
```

**Reply by jsilva74** [reply]
Hi,

Thanks for your reply. 

Yes, indeed this is pretty much what I'm using. The problem I'm facing is only with one of the tables Adonis uses to manage the migrations (adonis_schema) which is created automaticaly by Adonis.

Thank you.
